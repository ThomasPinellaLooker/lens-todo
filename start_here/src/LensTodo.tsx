import * as React from 'react'
import './LensTodo.css'

// Lens components
import {
  Card,
  CardContent,
  Flex,
  FlexItem,
  Link,
  theme,
  ThemeProvider,
} from 'looker-lens'

// Some useful utilities.
import
{
  TodoAppState,
  todoFromStr
} from './utils/todo_utils'

/**
 * Every React app is itself a component. Here we will define a basic container component,
 * an app state, and a composition of Lens and user defined components
 * to create a sweet Todo app.
 */
export default class LensTodo extends React.Component<{}, TodoAppState> {

  constructor(props: {}) {
    super(props)

    // Initial state of app. This can come from anything (e.g local storage, a server call, etc)
    // but for simplicity we will simply set here every time we start app.
    this.state = {
      showCompleted: false,
      todos: [
        todoFromStr('Install base app'),
        todoFromStr('Familiarize yourself with lens.looker.com'),
        todoFromStr('List todos'),
        todoFromStr('Create new todos'),
        todoFromStr('Complete todos'),
        todoFromStr('Filter todos'),
        todoFromStr('State management?'),
        todoFromStr('ðŸ’°Profit'),
      ],
    }
  }

  // The render function is called anytime a component's state or properties (props) change.
  public render() {

    return (
      <ThemeProvider theme={theme}>
        <Flex height="100%" justifyContent="center" alignItems="flex-start">
          <FlexItem>
            <Card className="todo-card" raised>
              <CardContent>
                It worked!  Add a <Link href="https://lens.looker.com/#/Components/Text?id=heading">title</Link> to the
                top of the card, and then Try and get the todos in your app component state to display here.  The todos
                then give a reasonable workflow for building out rest of app
              </CardContent>
            </Card>
          </FlexItem>
        </Flex>
      </ThemeProvider>
    )
  }
}
