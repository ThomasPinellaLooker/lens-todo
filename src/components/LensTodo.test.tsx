import { theme, ThemeProvider } from 'looker-lens';
import * as React from 'react'
import { create } from 'react-test-renderer';
import LensTodo from './LensTodo';

test('Default Todo', () => {
  const component = create(
    <ThemeProvider theme={theme}>
      <LensTodo />
    </ThemeProvider>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
