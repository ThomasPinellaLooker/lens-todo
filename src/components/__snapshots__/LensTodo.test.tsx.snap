// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form with one child 1`] = `
<div
  className="sc-EHOje empeAK sc-bdVaJa ladIop"
>
  <div
    className="sc-bZQynM dqpfBg sc-bdVaJa eWGIUr"
  >
    <div
      className="todo-card sc-htpNat kKxShc"
    >
      <div
        className="sc-bxivhb hnrkfU"
      >
        <div
          className="sc-EHOje empeAK sc-bdVaJa eXElVZ"
        >
          <h3
            className="sc-gqjmRU hHQcWf"
          >
            Get stuff done with Lens
          </h3>
        </div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="sc-EHOje empeAK sc-bdVaJa jkGTWO"
          >
            <div
              className="todo-input-container sc-bdVaJa eWGIUr"
            >
              <div
                className="sc-gzVnrw euKcLj"
              >
                <label
                  className="sc-htoDjs bUgUqE"
                  htmlFor={undefined}
                />
                <div
                  className="sc-EHOje empeAK sc-bdVaJa bZoazI"
                >
                  <div
                    className="sc-bZQynM dqpfBg sc-bdVaJa eWGIUr"
                  >
                    <input
                      className="sc-gZMcBi eLaHKN"
                      onChange={[Function]}
                      placeholder="Add a todo..."
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <button
              className="sc-bwzfXH hzkIez sc-bdVaJa cIzdnm"
              disabled={true}
            >
              Add it
            </button>
          </div>
        </form>
        <div
          className="sc-bdVaJa jZyCgK"
        >
          <ul
            className="sc-jTzLTM hUcpeU sc-bdVaJa fIZjAn"
          >
            <li
              className="sc-bdVaJa dmxNGQ"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="sc-EHOje empeAK sc-bdVaJa coSXsr"
              >
                <div
                  className="sc-EHOje empeAK sc-bdVaJa iFzqLF"
                >
                  <div
                    className="sc-bdVaJa hAeBdD"
                  >
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="active sc-kGXeez egbVtN"
                    size="3"
                  >
                    Setup skeleton project
                  </div>
                </div>
              </div>
            </li>
            <li
              className="sc-bdVaJa dmxNGQ"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="sc-EHOje empeAK sc-bdVaJa coSXsr"
              >
                <div
                  className="sc-EHOje empeAK sc-bdVaJa iFzqLF"
                >
                  <div
                    className="sc-bdVaJa hAeBdD"
                  >
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="active sc-kGXeez egbVtN"
                    size="3"
                  >
                    Get your bearings and have look around
                  </div>
                </div>
              </div>
            </li>
            <li
              className="sc-bdVaJa dmxNGQ"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="sc-EHOje empeAK sc-bdVaJa coSXsr"
              >
                <div
                  className="sc-EHOje empeAK sc-bdVaJa iFzqLF"
                >
                  <div
                    className="sc-bdVaJa hAeBdD"
                  >
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="active sc-kGXeez egbVtN"
                    size="3"
                  >
                    List todos
                  </div>
                </div>
              </div>
            </li>
            <li
              className="sc-bdVaJa dmxNGQ"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="sc-EHOje empeAK sc-bdVaJa coSXsr"
              >
                <div
                  className="sc-EHOje empeAK sc-bdVaJa iFzqLF"
                >
                  <div
                    className="sc-bdVaJa hAeBdD"
                  >
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="active sc-kGXeez egbVtN"
                    size="3"
                  >
                    Create new todos
                  </div>
                </div>
              </div>
            </li>
            <li
              className="sc-bdVaJa dmxNGQ"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="sc-EHOje empeAK sc-bdVaJa coSXsr"
              >
                <div
                  className="sc-EHOje empeAK sc-bdVaJa iFzqLF"
                >
                  <div
                    className="sc-bdVaJa hAeBdD"
                  >
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="active sc-kGXeez egbVtN"
                    size="3"
                  >
                    Complete todos
                  </div>
                </div>
              </div>
            </li>
            <li
              className="sc-bdVaJa dmxNGQ"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="sc-EHOje empeAK sc-bdVaJa coSXsr"
              >
                <div
                  className="sc-EHOje empeAK sc-bdVaJa iFzqLF"
                >
                  <div
                    className="sc-bdVaJa hAeBdD"
                  >
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="active sc-kGXeez egbVtN"
                    size="3"
                  >
                    Filter todos
                  </div>
                </div>
              </div>
            </li>
            <li
              className="sc-bdVaJa dmxNGQ"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="sc-EHOje empeAK sc-bdVaJa coSXsr"
              >
                <div
                  className="sc-EHOje empeAK sc-bdVaJa iFzqLF"
                >
                  <div
                    className="sc-bdVaJa hAeBdD"
                  >
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="active sc-kGXeez egbVtN"
                    size="3"
                  >
                    State management
                  </div>
                </div>
              </div>
            </li>
            <li
              className="sc-bdVaJa dmxNGQ"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="sc-EHOje empeAK sc-bdVaJa coSXsr"
              >
                <div
                  className="sc-EHOje empeAK sc-bdVaJa iFzqLF"
                >
                  <div
                    className="sc-bdVaJa hAeBdD"
                  >
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="active sc-kGXeez egbVtN"
                    size="3"
                  >
                    ðŸ’°Profit
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </div>
        <hr />
        <div
          className="sc-EHOje empeAK sc-bdVaJa eXElVZ"
        >
          <button
            className="sc-bwzfXH ekBYDi sc-bdVaJa cIzdnm"
            disabled={true}
            onClick={[Function]}
          >
            Show
             completed items
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
